### DISCUSSION TITLE
`paper.defaultConnector` - more than one connector

### DISCUSSION THREAD

**Discussion by GeorgiyZhuravlev** [clientIO/joint#2201] [Q&A] [answered]
### Introduction

Is it possible to use more than one connector type for paper's `defaultConnector`?
Or maybe "chain" them, somehow...
We're looking for a combination of `rounded` and `jumpover`, so, apply both to a link.

Looking at the type definition, seems not possible but, maybe, there is some kind of trick that could be used?
```
defaultConnector?: connectors.Connector | connectors.ConnectorJSON;
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
No, the `connectors` can not be chained.

You can use the `radius` option of [jumpover](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#connectors.jumpover) connector to make the corners rounded.

**Reply by GeorgiyZhuravlev** [reply]
Awesome! That's exactly the behavior we're looking for!
Missed that setting, sorry...
Thank you very much!

<img width="257" alt="image" src="https://github.com/clientIO/joint/assets/14068189/a333e285-b914-4b50-946f-6ca70c2162fb">

**Comment by GeorgiyZhuravlev** [comment]
It's a deeper hole than we first thought :)

Our clients tend to lay lines on top of each other. Which leads to this:

<img width="801" alt="image" src="https://github.com/clientIO/joint/assets/14068189/f903db77-f172-4f9e-9309-38363c623a0e">


here are the lines:

![_jumpover](https://github.com/clientIO/joint/assets/14068189/8df26622-0794-4ddc-8380-55599150c79d)


With the `rounded` connector lines looked great, but `jumpover` treats them as "intersected" and draws this small "arcs"
Could you give us a piece of advice on that, can we do anything with it?

**Reply by kumilingus** [reply]
I don't think we should create the jump if both paths have the same bearing after the jump (perhaps we can check and compare the tangents at the jump point).

It requires changes to the connector code in order to fix it. I can't think of any workaround other than making a copy of the connector and modifying it.
